test/suite/test_checkpoint_snapshot05.py

```diff
@@ -140,38 +141,43 @@ def test_checkpoint_snapshot(self):
             else:
                 cursor1.set_value(self.valueb + str(i))
             self.assertEqual(cursor1.update(), 0)
-
-        # Create a checkpoint thread
+        
+        # Commit the transaction during a checkpoint.
         done = threading.Event()
         ckpt = checkpoint_thread(self.conn, done, checkpoint_count_max=1)
         try:
             ckpt.start()
             
-            # Wait for checkpoint to start and acquire its snapshot before committing.
+            # Wait for checkpoint to acquire its snapshot executing the commit.
             ckpt_snapshot = 0
             while not ckpt_snapshot:
-                time.sleep(1)
-                stat_cursor = self.session.open_cursor('statistics:', None, None)
-                ckpt_snapshot = stat_cursor[stat.conn.checkpoint_snapshot_acquired][2]
-                stat_cursor.close()
+                with open_cursor(self.session, 'statistics:') as stat_cursor:
+                    ckpt_snapshot = stat_cursor[stat.conn.checkpoint_snapshot_acquired][2]
+                time.sleep(0.5)
 
             session1.commit_transaction()
             self.evict(self.uri, ds, self.nrows)
         finally:
             done.set()
             ckpt.join()
 
-        #Take a backup and restore it.
+        # Verify exactly one checkpoint is present: as required by the test assertions.
+        # Assertion is NOT about the code under test, so use 'assert'.
+        with open_cursor(self.session, 'statistics:') as stat_cursor:
+            started = stat_cursor[stat.conn.checkpoints][2]
+            skipped = stat_cursor[stat.conn.checkpoint_skipped][2]
+        assert (started - skipped) == 1
```

Comment:
If for some reason we perform 3 checkpoints and we skip 2, this will pass and I don't think this would be ok.
```suggestion
        assert stat_cursor[stat.conn.checkpoints][2] == 1
        assert stat_cursor[stat.conn.checkpoint_skipped][2] == 0
```

Reply:
This enforces a stricter condition: which I'm in favor of, but I was unsure whether it could be that strict. @kommiharibabu could you confirm if the expectation can be this strict?

Reply:
A checkpoint can be skipped only when it has nothing do. The test doesn't let the two checkpoints to be called due to the configuration of max number of checkpoints. The suggestion from Etienne is to make sure that we never skip the checkpoint because we expect a single checkpoint to be taken. Implementing the suggestion adds value to the expectation.